crates/apollo-router/src/configuration/otlp/mod.rs

```diff
@@ -33,11 +29,50 @@ pub struct Tracing {
     trace_config: Option<TraceConfig>,
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+pub enum Exporter {
+    #[cfg(feature = "otlp-grpc")]
+    Grpc(Option<TonicExporter>),
+    #[cfg(feature = "otlp-http")]
+    Http(Option<HttpExporter>),
+}
+
+impl Exporter {
+    pub fn exporter(&self) -> Result<opentelemetry_otlp::SpanExporterBuilder, ConfigurationError> {
+        match &self {
+            #[cfg(feature = "otlp-grpc")]
+            Exporter::Grpc(exporter) => Ok(exporter.clone().unwrap_or_default().exporter()?.into()),
+            #[cfg(feature = "otlp-http")]
+            Exporter::Http(exporter) => Ok(exporter.clone().unwrap_or_default().exporter()?.into()),
+        }
+    }
+
+    pub fn exporter_from_env() -> Result<opentelemetry_otlp::SpanExporterBuilder, ConfigurationError>
+    {
+        match std::env::var("ROUTER_TRACING").as_deref() {
+            #[cfg(feature = "otlp-http")]
+            Ok("http") => Ok(HttpExporter::exporter_from_env().into()),
+            #[cfg(feature = "otlp-grpc")]
+            Ok("tonic") => Ok(TonicExporter::exporter_from_env().into()),
+            Ok(val) => Err(ConfigurationError::InvalidEnvironmentVariable(format!(
+                "unrecognized value for ROUTER_TRACING: {}",
+                val
+            ))),
+            Err(e) => Err(ConfigurationError::MissingEnvironmentVariable(format!(
+                "could not read ROUTER_TRACING environment variable: {}",
+                e
+            ))),
+        }
```

Comment:
If both features are disabled there should be a warning somehow

Reply:
good point